hnn_core/gui/_viz_manager.py

```diff
@@ -273,11 +278,26 @@ def _dynamic_rerender(fig):
     fig.tight_layout()
 
 
-def _plot_on_axes(b, widgets_simulation, widgets_plot_type,
-                  target_simulations,
-                  spectrogram_colormap_selection, dipole_smooth,
-                  max_spectral_frequency, dipole_scaling, widgets, data,
-                  fig_idx, fig, ax, existing_plots):
+def create_plot_config(data, dipole_scaling, data_scaling, dipole_smooth,
+                       data_smooth, max_spectral_frequency,
+                       spectrogram_colormap_selection):
+
+    target_is_sim = data['net'] is not None
+    scaling = dipole_scaling.value if target_is_sim else data_scaling.value
+    smooth = dipole_smooth.value if target_is_sim else data_smooth.value
+    return {
+        "max_spectral_frequency": max_spectral_frequency.value,
+        "dipole_scaling": scaling,
+        "dipole_smooth": smooth,
+        "spectrogram_cm": spectrogram_colormap_selection.value
+    }
+
+
+def _plot_on_axes(b, simulations_widget, widgets_plot_type,
```

Comment:
Would you add the new `data_smooth` and `data_scaling` parameters to the doctoring?

Reply:
Done